### PR TITLE
tests with postgres

### PR DESCRIPTION
**Pull request by mszulik**
Check out the corresponding reusable workflow pullrequest: https://github.com/cybex-gmbh/github-workflows/pull/15

For some reason the postgres tests take very long on the github runner (~2 mins).

### Merge this only after merging https://github.com/cybex-gmbh/github-workflows/pull/15 and having set back the used reusable workflow branch to "@main".

### CODE REVIEW

File: phpunit-postgres.xml.dist [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  <logging>
+    <junit outputFile="build/report.junit.xml"/>
+  </logging>
+  <php>
+    <env name="PROTECTOR_PRIVATE_KEY" value="e195f1252346e31fe7b87e899f69a87d6fe99f38bd6e7c3cbdde411fcd9cc93e2c1d0ea6e0f8e207b38bef11bfcd5c0615c3cf4695876631b1da523a552b6022"/>
+    <env name="PROTECTOR_PUBLIC_KEY" value="2c1d0ea6e0f8e207b38bef11bfcd5c0615c3cf4695876631b1da523a552b6022"/>
+    <env name="PROTECTOR_ENCRYPTED_MESSAGE" value="c93e1c30857b0a5a36dddd11237ff65ef153144e0c400304cc80501da7e9b41a99c568fb34491629577ba4b5c0ec632e50067d61cf1ece27e1eabc"/>
+    <env name="PROTECTOR_DECRYPTED_MESSAGE" value="hello world"/>
+    <env name="PROTECTOR_AUTH_TOKEN" value="1|bxa5BBNo7lXxgtRY4DN775JdAtQSAlN4UfgRnFhz"/>
+    <env name="DB_CONNECTION" value="pgsql"/>
```
Comment (lupinitylabs) [inline phpunit-postgres.xml.dist:24]: Why do we need a separate phpunit config for postgres?
You usually set those environment variables in the github action instead.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,11 @@
 
 All notable changes to `protector` will be documented in this file.
 
+## v3.1.1 - 2024-12-11
```
Comment (mszulik) [inline CHANGELOG.md:5]: also add 3.1.2 which fixes postgres

**Review by lupinitylabs — APPROVED** [review]

File: src/Classes/AbstractPostgresSchemaStateProxy.php [hunk 3]
```diff
@@ -11,9 +11,9 @@
  */
 abstract class AbstractPostgresSchemaStateProxy extends PostgresSchemaState
 {
-    public function __construct(Connection $connection, protected PostgresSchemaState $schemaState, protected Protector $protector)
+    public function __construct(protected PostgresSchemaState $schemaState, protected Protector $protector)
     {
-        parent::__construct($connection);
+        parent::__construct($schemaState->connection);
```
[the remark this answers is not in the document]
Comment (mszulik) [reply src/Classes/AbstractPostgresSchemaStateProxy.php:16]: done.

File: CHANGELOG.md (hunk 2 above)
Comment (mszulik) [reply CHANGELOG.md:9]: done.

File: phpunit-postgres.xml.dist (hunk 1 above)
Comment (mszulik) [reply phpunit-postgres.xml.dist:24]: I refactored the workflow and am now setting those variables in the workflow itself.

**Review by gael-connan-cybex — APPROVED** [review]
tests run okay on github and locally

**Comment by gael-connan-cybex** [comment]
todo:
- replace branch refence for workflow
- add required tests to repo settings in github

**Pull request merged**